packages/daemon/src/service/pipeline.ts

```diff
@@ -218,10 +218,24 @@ export class PipelineService {
         throw new TypeError(`"${name}" plugin is required`);
       }
     };
-
+    const emitJobEvent = (jobStatus: PipelineStatus, step?: PluginTypeI, stepAction?: 'start' | 'end') => {
+      const jobEvent = new JobStatusChangeEvent(
+        jobStatus,
+        step,
+        stepAction
+      );
+      tracer.emit(jobEvent);
```

Comment:
In the JavaScript world, a emit function is more like to declare as: `emit(name, value)`.

Reply:
Or `push`?

Reply:
But here should use `emit`, may I ask you to tweak this?

Reply:
But the two parameters are binding, as we discussed above, we should depend on the typings to make it right.

Reply:
Just add an event type for the first argument.